core/modules/util/Error.h

```diff
@@ -74,6 +96,47 @@ class Error {
      * @return an output stream
      */
     friend std::ostream& operator<<(std::ostream &out, Error const& error);
+
+    int getCode() const {
+        return code;
+    }
+
+    void setCode(int code) {
+        this->code = code;
+    }
+
+    const std::string& getMsg() const {
+        return msg;
+    }
+
+    void setMsg(const std::string& msg) {
+        this->msg = msg;
+    }
+
+    int getStatus() const {
+        return status;
+    }
+
+    void setStatus(int status) {
+        this->status = status;
+    }
+
+    /** Check if current Object contains an actual error
+     *
+     *  by conversion code==util::ErrorCode::NONE
+     *  means that no error has been detected
+     *
+     * @return true if current object doesn't contain an actual error
+     */
+    bool isNone() {
+        return (code==util::ErrorCode::NONE);
+    }
+
+private:
+
+    int code;
+    std::string msg;
+    int status;
```

Comment:
No may be, but strong have to :) 
3-10. Private class variables and methods MUST have underscore prefix.